library/core/src/slice/mod.rs

```diff
@@ -3216,3 +3222,35 @@ impl<T> Default for &mut [T] {
         &mut []
     }
 }
+
+#[unstable(feature = "slice_pattern", reason = "stopgap trait for slice patterns", issue = "56345")]
+/// Patterns in slices - currently, only used by `strip_prefix` and `strip_suffix`.  At a future
+/// point, we hope to generalise `core::str::Pattern` (which at the time of writing is limited to
+/// `str`) to slices, and then this trait will be replaced or abolished.
+pub trait SlicePattern {
+    /// The element type of the slice being matched on.
+    type Item;
+
+    /// Currently, the consumers of `SlicePattern` need a slice.
+    fn as_slice(&self) -> &[Self::Item];
+}
+
+#[stable(feature = "slice_strip", since = "1.50.0")]
```

Comment:
Stability attributes don't work on trait impls, I think you can just remove them.

Reply:
They are being rendered by rustdoc, e.g. https://doc.rust-lang.org/core/num/struct.NonZeroU128.html#impl-TryFrom%3CNonZeroU128%3E-1